src/go/rpk/pkg/kafka/utils.go

```diff
@@ -0,0 +1,256 @@
+			return bs
+		}
+		// Otherwise, try to find an existing config file.
+		conf, err := configuration()
+		if err != nil {
+			log.Trace(
+				"Couldn't read the config file." +
+					" Assuming 127.0.0.1:9092",
+			)
+			log.Debug(err)
+			return []string{"127.0.0.1:9092"}
+		}
+
+		// Add the seed servers' Kafka addrs.
+		if len(conf.Redpanda.SeedServers) > 0 {
+			for _, b := range conf.Redpanda.SeedServers {
+				addr := fmt.Sprintf(
+					"%s:%d",
+					b.Host.Address,
+					conf.Redpanda.KafkaApi.Port,
+				)
+				bs = append(bs, addr)
+			}
+		}
+		// Add the current node's Kafka addr.
+		selfAddr := fmt.Sprintf(
+			"%s:%d",
+			conf.Redpanda.KafkaApi.Address,
+			conf.Redpanda.KafkaApi.Port,
+		)
+		bs = append(bs, selfAddr)
+		log.Debugf(
+			"Using brokers from config: %s",
+			strings.Join(bs, ", "),
+		)
+		return bs
+	}
+}
+
+func CreateProducer(
+	brokers func() []string, configuration func() (*config.Config, error),
+) func(bool, int32) (sarama.SyncProducer, error) {
+	return func(jvmPartitioner bool, partition int32) (sarama.SyncProducer, error) {
+		conf, err := configuration()
+		if err != nil {
+			return nil, err
+		}
+		cfg, err := LoadConfig(conf)
+		if err != nil {
+			return nil, err
+		}
+		if jvmPartitioner {
+			cfg.Producer.Partitioner = kafkautil.NewJVMCompatiblePartitioner
+		}
+
+		if partition > -1 {
+			cfg.Producer.Partitioner = sarama.NewManualPartitioner
+		}
+
+		return sarama.NewSyncProducer(brokers(), cfg)
+	}
+}
+
+func PublishMessage(
+	message []byte,
+	key string,
+	producer sarama.SyncProducer,
+	topic string,
+	header []sarama.RecordHeader,
+) error {
+
+	k := sarama.StringEncoder(key)
+	ts := time.Now()
+
+	msg := &sarama.ProducerMessage{
+		Topic:		topic,
+		Key:		k,
+		Timestamp:	ts,
+		Value:		sarama.ByteEncoder(message),
+		Headers:	header,
+	}
+
+	retryConf := DefaultConfig().Producer.Retry
+	part, offset, err := RetrySend(
+		producer,
+		msg,
+		uint(retryConf.Max),
+		retryConf.Backoff,
+	)
+	if err != nil {
+		log.Error("Failed to send record")
+		return err
+	}
+
+	log.Infof(
+		"Sent record to partition %d at offset %d with timestamp %v.",
+		part,
+		offset,
+		ts,
+	)
+	return nil
+}
+
+func CreateClient(
+	fs afero.Fs,
+	brokers func() []string,
+	configuration func() (*config.Config, error),
+) func() (sarama.Client, error) {
+	return func() (sarama.Client, error) {
+		conf, err := configuration()
+		if err != nil {
+			return nil, err
+		}
+		bs := brokers()
+		return InitClientWithConf(conf, bs...)
+	}
+}
+
+func CreateAdmin(
+	fs afero.Fs,
+	brokers func() []string,
+	configuration func() (*config.Config, error),
+) func() (sarama.ClusterAdmin, error) {
+	return func() (sarama.ClusterAdmin, error) {
+		conf, err := configuration()
+		if err != nil {
+			return nil, err
+		}
+		cfg, err := LoadConfig(conf)
+		if err != nil {
+			return nil, err
+		}
+		return sarama.NewClusterAdmin(brokers(), cfg)
+	}
+}
+
+func ExistingTopic(admin sarama.ClusterAdmin, topic string) (bool, error) {
```

Comment:
Where is this function used?

Reply:
on `deploy.go` and `remove.go` https://github.com/vectorizedio/redpanda/pull/392/files#diff-3f51d0d2a86c5c576d217ba895b88d5637df681634db619f33692f9d6110e79eR92

Reply:
Then they should go in `cli/cmd/wasm/common/common.go`, since they're only used there.

Reply:
https://github.com/vectorizedio/redpanda/pull/392/files#diff-3f51d0d2a86c5c576d217ba895b88d5637df681634db619f33692f9d6110e79eR92

https://github.com/vectorizedio/redpanda/pull/392/files#diff-ab6c2e9d46e0e8d2f0eb6217c44acf868bd91f5eed3700f1967940c0f580b2d6R53

Reply:
I created `common` into `/wasm` because we just it in wasm context